src/Talis/Persona/Client/ComplexKeyCacheStorage.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Cache HTTP requests by defining the cache key with the request headers
+ */
+namespace Talis\Persona\Client;
+use Guzzle\Plugin\Cache\DefaultCacheStorage;
+use Guzzle\Http\Message\RequestInterface;
+
+/**
+ * Define a DefaultCacheStorage implementation that also uses the
+ * request HTTP headers as the cache key
+ */
+class ComplexKeyCacheStorage extends DefaultCacheStorage
```

Comment:
All of the keys are using md5 to create a hash. What differs here is the use of the headers within the key.